libraries/eosiolib/capi/eosio/types.h

```diff
@@ -32,15 +32,25 @@ typedef uint64_t capi_name;
 /**
  * EOSIO Public Key. K1 and R1 keys are 34 bytes.  Newer keys can be variable-sized
  */
-struct __attribute__((deprecated("newer public key types cannot be represented as a fixed size structure", "char[]")))
+struct 
+#ifdef __GNUC__
+__attribute__((deprecated("newer public key types cannot be represented as a fixed size structure, use char[] instead")))
```

Comment:
Why have the `ifdef`?

Reply:
clang's `deprecated` has two parameters (one for type name, another for suggested replacement).
gcc has one parameter of single string

Reply:
clang's feature is nicer, though it's optional, it seems worth keeping.